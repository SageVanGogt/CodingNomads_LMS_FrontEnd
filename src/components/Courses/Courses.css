.courses_container {
  display: -ms-flexbox;
  display: flex;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -ms-flex-pack: distribute;
  justify-content: space-around;
  -ms-flex-align: space-around;
  align-items: space-around;
  margin-left: 200px;
  width: calc(100vw - 200px);
  padding-top: 100px;
  background-color: #efefef;
  min-height: 100vh; }

.course_cards {
  display: -ms-flexbox;
  display: flex;
  -ms-flex-pack: center;
  justify-content: center;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 80%;
  background-color: #efefef;
  border-radius: 5px;
  height: fit-content; }

.new_course_button {
  position: fixed;
  top: 20px;
  right: 20px;
  width: 120px;
  height: 40px;
  background-color: #4885a2;
  border-radius: 5px;
  font-weight: 800;
  font-size: .9rem;
  color: white;
  cursor: pointer; }
