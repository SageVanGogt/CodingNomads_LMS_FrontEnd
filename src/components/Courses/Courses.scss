.courses_container {
  display: flex;
  flex-wrap: wrap;
  margin-left: 225px;
  justify-content: space-around;
}