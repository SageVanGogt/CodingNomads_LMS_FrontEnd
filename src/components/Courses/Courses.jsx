import React, { Component } from 'react';
import { mockCourses } from '../../mockData/mockCourses';
import './Courses.css';
import CourseCard from '../CourseCard/CourseCard';

export class Courses extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const courses = mockCourses.map(course => <CourseCard {...course} key={'course' + course.id}/>)

    return ( 
      <div className='courses_container'>
        {courses}
      </div>
    );
  }
}