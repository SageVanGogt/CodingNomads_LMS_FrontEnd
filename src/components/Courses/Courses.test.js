import React from 'react';
import { shallow } from 'enzyme';
import { Courses } from './Courses';

describe('TaskCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Courses />, { disableLifecycleMethods: true });
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
