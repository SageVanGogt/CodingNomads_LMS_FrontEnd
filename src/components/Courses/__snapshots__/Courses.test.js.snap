// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Courses />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="new_course_button"
          onClick={[Function]}
>
          New Course +
</button>,
        <section
          className="course_cards"
>
          <img
                    height="200"
                    src="/loading1.gif"
                    width="200"
          />
</section>,
      ],
      "className": "courses_container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "New Course +",
          "className": "new_course_button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "New Course +",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            height="200"
            src="/loading1.gif"
            width="200"
/>,
          "className": "course_cards",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "200",
            "src": "/loading1.gif",
            "width": "200",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="new_course_button"
            onClick={[Function]}
>
            New Course +
</button>,
          <section
            className="course_cards"
>
            <img
                        height="200"
                        src="/loading1.gif"
                        width="200"
            />
</section>,
        ],
        "className": "courses_container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "New Course +",
            "className": "new_course_button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "New Course +",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              height="200"
              src="/loading1.gif"
              width="200"
/>,
            "className": "course_cards",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "height": "200",
              "src": "/loading1.gif",
              "width": "200",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
